docs/content/guides/migrations/coexistence-of-airflow-and-dagster.mdx

```diff
@@ -0,0 +1,97 @@
+---
+title: "Coexistence of Airflow and Dagster | Dagster Docs"
+description: "Learn how to leverage the features of Dagster and Airflow together."
+---
+
+# Coexistence of Airflow and Dagster
+
+Dagster can act as a single entry point to all orchestration platforms in use at your organization. By injecting a small amount of code into your existing pipelines, you can report events to Dagster, where you can then visualize the full lineage of pipelines. This can be particularly useful if you have multiple Apache Airflow environments, and hope to build a catalog and observation platform through Dagster.
+
+### Emitting materialization events from Airflow to Dagster
+
+Imagine you have a large number of pipelines written in Apache Airflow, and you wish to introduce Dagster into your stack. You can continue to run your existing pipelines, while building new pipelines in Dagster that are tightly integrated to your legacy systems by using custom Airflow operators.
+
+To do this, we will define a `DagsterAssetOperator` operator that can be appended to the end of your Airflow DAG to indicate the processing of the pipeline has concluded. The HTTP endpoint of Dagster server, the `asset_key` and additional metadata and descriptions are to be specified to inform Dagster of the materialization.
+
+```python
+from typing import Dict, Optional
+
+from airflow.models import BaseOperator
+from airflow.utils.decorators import apply_defaults
+import requests
+
+class DagsterAssetOperator(BaseOperator):
+    @apply_defaults
+    def __init__(
+        self,
+        dagster_webserver_host: str,
+        asset_key: str,
+        metadata: Optional[Dict] = None,
+        description: Optional[str] = None,
+        *args,
+        **kwargs,
+    ):
+        super().__init__(*args, **kwargs)
+        self.dagster_webserver_host = dagster_webserver_host
+        self.asset_key = asset_key
+        self.metadata = metadata or {}
+        self.description = description
+
+    def execute(self, context):
+        url = f"http://{self.dagster_webserver_host}/report_asset_materialization/{self.asset_key}"
+        payload = {"metadata": self.metadata, "description": self.description}
+        headers = {"Content-Type": "application/json"}
+
+        response = requests.post(url, json=payload, headers=headers)
+        response.raise_for_status()
+
+        self.log.info(
+            f"Reported asset materialization to Dagster. Response: {response.text}"
+        )
+```
+
+Then, we can append this to our Airflow DAG to indicate that a pipeline has run to success.
+
+```python
+dagster_op = DagsterAssetOperator(
+    task_id="report_dagster_asset_materialization",
+    dagster_webserver_host="localhost:3000",
```

Comment:
We may want to update the example to include `dagster_webserver_host` and `dagster_webserver_port` to match our current implementation in the CLI, see [docs](https://docs.dagster.io/_apidocs/cli#dagster-webserver) and [code](https://github.com/dagster-io/dagster/blob/1.7.10/python_modules/dagster-webserver/dagster_webserver/cli.py#L181-L197).

Reply:
This is helpful, thanks. Is this similar to what you envisioned?

https://github.com/dagster-io/dagster/pull/22638/commits/6b80ebf3b2bf4289ef18c4e7b0764350e95b1c23

Reply:
Yes! Maybe having two separate parameters would be ideal?

```python
class DagsterAssetOperator(BaseOperator):
    ...
    
    def execute(self, context):
        url = f"http://{dagster_webserver_host}:{dagster_webserver_port}/report_asset_materialization/{self.asset_key}"
        ...

dagster_webserver_host = os.environ.get("DAGSTER_WEBSERVER_HOST", "localhost")
dagster_webserver_port = os.environ.get("DAGSTER_WEBSERVER_PORT", "3000")

dagster_op = DagsterAssetOperator(
    task_id="report_dagster_asset_materialization",
    dagster_webserver_host=dagster_webserver_host,
    dagster_webserver_port=dagster_webserver_port,
    asset_key="example_external_airflow_asset",
    metadata={"airflow/tag": "example", "source": "external"},
)
```

Reply:
Updated, that's must cleaner, ty!